src/components/CardChapter.tsx

```diff
@@ -20,9 +20,10 @@ export const CardChapter = ({chapter}: {chapter: chapterType}) => {
       )}
       <Text style={styles.textColor}> {chapter.title}</Text>
       <TouchableOpacity
+        disabled={chapterShow.includes(chapter.id)}
         onPress={() => {
-          useAppStore.setState({chapterShow: [...chapterShow, chapter.id]});
-          setRefresh(!refresh);
+          toggleChapterStatus(chapter.id);
+          setRefresh(prev => !prev);
```

Comment:
This is not a reactive pattern. React is supposed to know to update the UI based on state changes, without you having to add an extra state to trigger the re-render.

Reply:
Maybe we need pass refresh to store for refresh or in componant parents for refresh Flatlist.